lib/rubocop/cop/lint/rand_one.rb

```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for `rand(1)` calls.
+      # Such calls always return `0`.
+      #
+      # @example
+      #
+      #   @bad
+      #   rand 1
+      #   Kernel.rand(-1)
+      #   rand 1.0
+      #   rand(-1.0)
+      #
+      #   @good
+      #   0
+      class RandOne < Cop
+        MSG = '`%s` always returns `0`. Perhaps you meant `rand(2)` or `rand`?'
+        ONES = [1, -1, 1.0, -1.0]
+
+        def_node_matcher :rand_call, <<-PATTERN
+          (send {(const nil :Kernel) nil} :rand {(int $_) (float $_)})
+        PATTERN
```

Comment:
Alternatively, you could include the literals here in the pattern. I'm not sure if we support float literals in node patterns... but I could add it if necessary.